locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,427 @@
+
+Possible vector attacks:
+
+* Typosquatting attacks
+* Lockfile poisoning
+* Compromised maintainers
+* Malicious Packages
+* Dependency Confusions
+
+**Mitigation**
+
+* Using native modules with prebuilds to avoid native buildings, and verifying SHAs
+* Prevent npm from executing arbitrary scripts / Disabling npm postinstall
+* Pin dependency versions
+* Use lockfiles, which pin every dependency (direct and transitive).
+  * Use [Mitigations for lockfile poisoning][].
+* Automate checks for new vulnerabilities using CI, with tools like npm-audit.
+* Use [`npm ci`][] instead of `npm install`.
+This enforces the lockfile so that inconsistencies between it and the
+_package.json_ file cause an error (instead of silently ignoring the lockfile
+in favor of _package.json_).
+* Use [`npm audit`][] in CI to frequently scan for vulnerabilities in your
+dependencies.
+* Carefully check the _package.json_ file for errors/typos in the names of the
+dependencies.
+
+### Memory Access Violation (CWE-284)
+
+Memory-based or heap-based attacks depends on a combination of memory management
+errors and an exploitable memory allocator.
+Like all runtimes, Node.js is vulnerable to these attacks if your projects runs
+on a shared machine.
+Using a secure heap is useful for preventing sensitive information from leaking
+due to pointer overruns and underruns.
+
+Unfortunately, secure heap is not available on Windows.
+More information can be found on Node.js [secure-heap documentation][].
+
+**Mitigation**
+
+* Use `--secure-heap=n` depending on your application where _n_ is the allocated
+maximum byte size.
+* Do not run your production app in a shared machine.
+
+### Monkey Patching (CWE-349)
+
+Monkey patching refers to the modification of properties in runtime aiming to
+change the existing behavior. Example:
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+```
+
+This is **not** considered a vulnerability, it falls under the following
+statement: _“Node.js trusts the code being run”_.
+
+**Mitigation**
+
+The `--frozen-intrinsics` flag enables **experimental** frozen intrinsics, which
+means all the built-in JavaScript objects and functions are recursively frozen.
+Therefore, the following snippet **will not** override the default behavior of
+`Array.prototype.push`
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+
+// Uncaught:
+// TypeError <Object <Object <[Object: null prototype] {}>>>:
+// Cannot assign // to read only property 'push' of object ''
+```
+
+However, it’s important to mention you can still define new globals and replace
+existing globals using `globalThis`
+```console
+> globalThis.foo = 3; foo; // you can still define new globals
+3
+> globalThis.Array = 4; Array; // However, you can also replace existing globals
+4
+```
```

Comment:
You can always freeze the global to avoid potential issues from attackers doing this. But this seems potentially like a bug in the flag?

Reply:
yes, but that has some bad effects due to what is caused by the "Override Mistake" if you don't do some repairs that frozen-intrinsics that we probably don't want to get into. Most likely we should just fix `--frozen-intrinsics`. I'd note that private fields can be attached / WeakMap can be used with frozen stuff still if they need that.

Reply:
I'll mention the `Object.freeze(globalThis)` until we don't fix the --frozen-intrinsics then.